analyzers/src/SonarAnalyzer.CFG/Sonar/CSharpControlFlowGraphBuilder.cs

```diff
@@ -1286,6 +1286,36 @@ private Block BuildPatternExpression(PatternSyntaxWrapper patternSyntaxWrapper,
                 currentBlock.ReversedInstructions.Add(patternSyntaxWrapper);
                 return currentBlock;
             }
+            else if (ParenthesizedPatternSyntaxWrapper.IsInstance(patternSyntaxWrapper))
+            {
+                var parenthesized = (ParenthesizedPatternSyntaxWrapper)patternSyntaxWrapper;
+                return BuildPatternExpression(parenthesized.Pattern, currentBlock);
+            }
+            else if (BinaryPatternSyntaxWrapper.IsInstance(patternSyntaxWrapper))
+            {
+                // Unsupported
+                return currentBlock;
+            }
+            else if (UnaryPatternSyntaxWrapper.IsInstance(patternSyntaxWrapper))
+            {
+                // Unsupported
+                return currentBlock;
+            }
+            else if (VarPatternSyntaxWrapper.IsInstance(patternSyntaxWrapper))
+            {
+                // Unsupported
+                return currentBlock;
+            }
+            else if (RelationalPatternSyntaxWrapper.IsInstance(patternSyntaxWrapper))
+            {
+                // Unsupported
+                return currentBlock;
+            }
+            else if (ListPatternSyntaxWrapper.IsInstance(patternSyntaxWrapper))
+            {
+                // Unsupported
+                return currentBlock;
+            }
```

Comment:
This whole part is deprecated and we don't want to add any new features here. Ever.
Return `currentBlock` in `else` everytime just to prevent failing.

Reply:
I kept the current approach of the code base of throwing and kept the existing tests and added two more for the new "unsupported" syntaxKinds. I think we might produce an unbalanced stack if we don't throw but instead return the state.

Reply:
This is a CFG Builder -> returning `currentBlock` means no-op.

Let's never throw here. sonar-security C# FE will benefit from the behavior too.

Reply:
> returning currentBlock means no-op.

This is what concerns me. The engine would continue as if nothing happened. If we throw instead, we basically stop the analysis of the current method, right? We already throw for e.g. VarPattern, And/Or/Not pattern, and others. I followed their example in the latest version of the PR (this discussion is a bit outdated).

Reply:
This is not the engine. This is CFG builder. It will throw. SE should not run. And we don't care.

I'll close this one, as we should deal with the other comment instead:
https://github.com/SonarSource/sonar-dotnet/pull/6768#discussion_r1115651491